admin/config-ui/fields/class-field-environment.php

```diff
@@ -27,6 +27,69 @@ public function __construct() {
 	 * @param WPSEO_Configuration_Options_Adapter $adapter Adapter to register lookup on.
 	 */
 	public function set_adapter( WPSEO_Configuration_Options_Adapter $adapter ) {
-		$adapter->add_yoast_lookup( $this->get_identifier(), 'wpseo', 'environment_type' );
+		$adapter->add_custom_lookup(
+			$this->get_identifier(),
+			array( $this, 'get_data' ),
+			array( $this, 'set_data' )
+		);
+	}
+
+	/**
+	 * Gets the option that is set for this field.
+	 *
+	 * @return string The value for the environment_type wpseo option.
+	 */
+	public function get_data() {
+		$option = WPSEO_Options::get_option( 'wpseo' );
+
+		return $option['environment_type'];
+	}
+
+	/**
+	 * Set new data.
+	 *
+	 * @param string $environment_type The site's environment type.
+	 *
+	 * @return bool Returns whether the value is successfully set.
+	 */
+	public function set_data( $environment_type ) {
+		$set_wp_indexation_successful = false;
+		$option                       = WPSEO_Options::get_option( 'wpseo' );
+
+		if ( $option['environment_type'] !== $environment_type ) {
+			$option['environment_type'] = $environment_type;
+			update_option( 'wpseo', $option );
+			$set_wp_indexation_successful = $this->set_indexation( $environment_type );
```

Comment:
I would do: 
```
if ( !  $this->set_indexation( $environment_type ) ) {
   return false;
}
```

If you do it this way you can remove `$set_wp_indexation_successful`